go/common/amount/amount.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package amount
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/holiman/uint256"
+)
+
+// Amount is a 256-bit unsigned integer used for token values like balances.
+type Amount struct {
+	internal uint256.Int
+}
+
+// NewAmount creates a new U256 Amount from up to 4 uint64 arguments. The
+// arguments are given in the Big Endian order. No argument results in a value of zero.
+// The constructor panics if more than 4 arguments are given.
+func NewAmount(args ...uint64) Amount {
```

Comment:
Maybe, since we have a dedicated package now, we can shorten the name of this function to `New`.
This would match `big.New` or `uint256.New`.